process-create-data-repo-request

```diff
@@ -0,0 +1,316 @@
+        parser.error("Too many arguments")
+
+    GH_TOKEN = open(expanduser("~/.github-token")).read().strip()
+
+    issue_id = int(args[0])
+    gh = Github(login_or_token=GH_TOKEN)
+    api_rate_limits(gh)
+    cmssw_repo_name = GH_CMSSW_ORGANIZATION + "/" + GH_CMSSW_REPO
+    cmssw_repo = gh.get_repo(cmssw_repo_name)
+    issue = cmssw_repo.get_issue(issue_id)
+    print("API Rate Limit")
+    print("Limit, Remaining: ", gh.rate_limiting)
+    print("Reset time (GMT): ", datetime.fromtimestamp(gh.rate_limiting_resettime))
+
+    # 0. Is this issue closed?
+    if issue.state == "closed":
+        print("Issue closed, ignoring.")
+        exit(0)
+
+    # 1. Is this a pull request?
+    if issue.pull_request:
+        print("This is a pull request, ignoring.")
+        exit(0)
+
+    title_match = re.match(CREATE_REPO, issue.title)
+
+    # 2. Is this issue meant to create a new cms-data repo?
+    if not title_match:
+        print("This issue is not for creating a repo, ignoring.")
+        print(issue.title)
+        exit(0)
+
+    category_name = title_match.group(1)
+    package_name = title_match.group(2)
+
+    print(category_name + "/" + package_name)
+
+    # 3. Does the requested repository already exist?
+    repo = None
+    try:
+        repo = gh.get_organization("cms-data").get_repo(
+            category_name + "-" + package_name
+        )
+    except github.UnknownObjectException:
+        pass
+
+    comments = [c for c in issue.get_comments()]
+
+    if repo:
+        post_message(issue, EXISTS_MSG.format(repo=repo.url), comments)
+        issue.edit(state="closed")
+        exit(0)
+
+    # Figure out who must approve the action
+    data_categs = []
+    for cat, pkgs in CMSSW_CATEGORIES.items():
+        for pkg in pkgs:
+            if re.match(pkg + ".*", category_name + "/" + package_name):
+                data_categs.append(cat)
+                break
+
+    if not data_categs:
+        post_message(
+            issue,
+            INVALID_REQUEST_MSG.format(package=category_name + "/" + package_name),
+            comments,
+        )
+        exit(0)
+
+    # Get the status of this issue.
+    status = get_issue_status(issue)
+    print("Status: %s \n" % status)
+    print("Issue labels:", labels)
+
+    if status == NEW_ISSUE:
+        post_message(issue, "assign " + ",".join(data_categs), comments)
+        set_comment_emoji(comments[0].id, issue.repository.full_name)
+        exit(0)
+
+    if status == PENDING_APPROVAL:
+        print("Request not approved yet")
+        exit(0)
+
+    if status == FULLY_SIGNED:
+        org = gh.get_organization("cms-data")
+        new_repo = org.create_repo(
+            category_name + "-" + package_name,
+            "Data files for " + category_name + "/" + package_name,
+            has_wiki=False,
+            has_projects=False,
+        )
+
+        post_message(issue, COMPLETE_MSG.format(new_repo.url), comments)
+        issue.edit(state="closed")
```

Comment:
for all the places where this script changes state of the issue please add a dry run check. e.g adding comment, adding emoji, creating data repo etc.